frontend/src/components/FacilityBooking/BookingSection.tsx

```diff
@@ -1,60 +1,107 @@
-  const defaultTimePosition = 16 //4pm (can range from 0 to 23 - length of timeBlocks)
+  const location = useLocation<ViewBookingLocationState>()
+  const { timeBlocks, selectedBlockTimestamp, selectedStartTime, selectedEndTime } = useSelector(
+    (state: RootState) => state.facilityBooking,
+  )
+  const { clickedDate } = useSelector((state: RootState) => state.calendar)
+
+  const defaultTimePosition = 16 // 4pm (can range from 0 to 23 - length of timeBlocks)
+
+  const dateRowStartDate = location.state.dateRowStartDate
 
   useEffect(() => {
-    dispatch(getTimeBlocks())
+    dispatch(getTimeBlocks(clickedDate))
   }, [])
 
   useEffect(() => {
-    if (selectedStartTime !== -1) {
-      let isAfterOccupied = false
-      const newTimeblocks: TimeBlock[] = timeBlocks.map((entry) => {
-        let type = entry.type
-        if (entry.timestamp === selectedBlockTimestamp) {
-          type = TimeBlockType.SELECTED
-        } else if (entry.timestamp < selectedBlockTimestamp) {
-          type = type === TimeBlockType.AVAILABLE ? TimeBlockType.UNAVAILABLE : type
-        } else {
-          if (entry.type === TimeBlockType.OCCUPIED) {
-            isAfterOccupied = true
-          }
-          if (isAfterOccupied) {
-            type = type === TimeBlockType.AVAILABLE ? TimeBlockType.UNAVAILABLE : type
-          }
-        }
-        return { ...entry, type }
-      })
-      dispatch(setTimeBlocks(newTimeblocks))
-    }
+    updateTimeBlocks()
   }, [selectedBlockTimestamp])
 
-  function setSelectedBlock(selectedTimestamp: number) {
-    setSelectedBlockTimestamp(selectedTimestamp)
-    if (selectedStartTime === -1) {
-      dispatch(setStartEndTime(selectedTimestamp))
+  const assignType = (entry: TimeBlock): TimeBlockType => {
+    if (entry.timestamp === selectedBlockTimestamp) return TimeBlockType.SELECTED
+    if (entry.timestamp < selectedBlockTimestamp)
+      return entry.type === TimeBlockType.AVAILABLE ? TimeBlockType.UNAVAILABLE : entry.type
+    return entry.type
+  }
+
+  const updateTimeBlocks = () => {
+    if (selectedStartTime === 0) return
+    const newTimeblocks: TimeBlock[] = timeBlocks.map((entry) => {
+      const type = assignType(entry)
+      return { ...entry, type }
+    })
+    dispatch(setTimeBlocks(newTimeblocks))
+  }
+
+  const goToBookingPage = () => {
+    history.push({
+      pathname: PATHS.CREATE_FACILITY_BOOKING + facilityId,
+      state: {
+        dateRowStartDate: dateRowStartDate,
+      },
+    })
+  }
+
+  const setSelectedBlock = (selectedTimestamp: number) => {
+    dispatch(setSelectedBlockTimestamp(selectedTimestamp))
+
+    if (selectedStartTime === 0) {
+      if (selectedEndTime === 0) {
+        // selecting start time first before selecting end time
+        dispatch(setSelectedStartTime(selectedTimestamp))
+      } else {
+        // reselecting start time only, then go back to booking page
+        if (selectedTimestamp > selectedEndTime) {
```

Comment:
might want to confirm if this is a use case we should allow? I think we can afford:
1. reselect start and end time (user clicks start time field)
2. reselect end time only (user clicks end time field)

Reply:
checked with designers, they intend reselect start time only as well